Tests/BSONTests/BSONCorpusTests.swift

```diff
@@ -129,97 +160,96 @@ final class BSONCorpusTests: BSONTestCase {
+                        expect(try canonicalEncoder.encode(docFromDB))
+                            .to(cleanEqual(test.canonicalExtJSON))
+
+                        // native_to_relaxed_extended_json( bson_to_native(dB) ) = rEJ (if rEJ exists)
+                        // (Not in spec yet, might be added in DRIVERS-1355)
+                        if let rEJ = test.relaxedExtJSON {
+                            expect(try relaxedEncoder.encode(docFromDB))
+                                .to(cleanEqual(rEJ), description: test.description)
+                        }
+                    }
 
-                    //     // native_to_bson( json_to_native(dEJ) ) = cB (unless lossy)
-                    //     if !lossy {
-                    //         expect(try Document(fromJSON: dEJ).rawBSON).to(equal(cBData))
-                    //     }
-                    // }
+                    // for dEJ input (if it exists):
+                    if let dEJ = test.degenerateExtJSON, let dEJData = dEJ.data(using: .utf8) {
+                        // native_to_canonical_extended_json( json_to_native(dEJ) ) = cEJ
+                        expect(try canonicalEncoder.encode(try decoder.decode(BSON.self, from: dEJData)))
+                            .to(cleanEqual(test.canonicalExtJSON), description: test.description)
+                        // native_to_bson( json_to_native(dEJ) ) = cB (unless lossy)
+                        if !lossy {
+                            try expect(try decoder.decode(BSONDocument.self, from: dEJData))
+                                .to(sortedEqual(BSONDocument(fromBSON: cBData)), description: test.description)
+                        }
+                    }
 
                     // for rEJ input (if it exists):
-                    // if let rEJ = test.relaxedExtJSON {
-                    //     // native_to_relaxed_extended_json( json_to_native(rEJ) ) = rEJ
-                    //     expect(try Document(fromJSON: rEJ).extendedJSON).to(cleanEqual(rEJ))
-                    // }
+                    if let rEJ = test.relaxedExtJSON, let rEJData = rEJ.data(using: .utf8) {
+                        // native_to_relaxed_extended_json( json_to_native(rEJ) ) = rEJ
+                        expect(try relaxedEncoder.encode(try decoder.decode(BSON.self, from: rEJData)))
+                            .to(cleanEqual(rEJ), description: test.description)
+                    }
                 }
             }
 
             if let parseErrorTests = testFile.parseErrors {
                 continue // TODO: EXT JSON support required
                 for test in parseErrorTests {
+                    guard !shouldSkip(testFile.description, test.description) else {
+                        continue
+                    }
                     let description = "\(testFile.description)-\(test.description)"
-
+                    guard let testData = Data(hexString: test.string) else {
```

Comment:
this `testData` portion should be moved into the "top level document" case, since for others (e.g. decimal128) `test.string` doesn't need to be converted to `Data`. Also, instead of using this `hexString` initializer, we need to be using the regular `.data(using: .utf8)`, since the string isn't a hex string but is extended JSON.

Reply:
okay